src/main/webapp/transcript.js

```diff
@@ -0,0 +1,71 @@
+
+const TRANSCRIPT_CONTAINER = 'transcript-container';
+
+/**
+ * Sends a POST request to the transcript.
+ */
+function sendPostToTranscript() {
+  // TODO: Update with user input.
+  const paramsString = 'id=123456789&video=3ymwOvzhwHs';
+  const params = new URLSearchParams(paramsString);
+  fetch('/transcript', {method: 'POST', body: params})
+      .then(fetchTranscriptLines(paramsString));
+}
+
+/**
+ * Fetches the transcript lines from \transcript.
+ *
+ * <p>{@code paramsString} indicates the video ID
+ * to fetch the transcript from.
+ */
+function fetchTranscriptLines(paramsString) {
+  fetch(
+      '/transcript' +
+      '?' + paramsString)
+      .then((response) => response.json())
+      .then((transcriptLines) => {
+        addMultipleTranscriptLinesToDom(transcriptLines);
+      });
+}
+
+/**
+ * Adds {@code transcriptLines} to the DOM as list elements.
+ */
+function addMultipleTranscriptLinesToDom(transcriptLines) {
+  const transcriptContainer = document.getElementById(TRANSCRIPT_CONTAINER);
+  if (transcriptContainer.firstChild) {
+    transcriptContainer.removeChild(transcriptContainer.firstChild);
+  }
+  const ulElement = document.createElement('ul');
+  transcriptContainer.appendChild(ulElement);
+
+  transcriptLines.forEach((transcriptLine) => {
+    appendTextToList(transcriptLine, ulElement);
+  });
+}
+
+/**
+ * Creates an <li> element containing {@code transcriptLine}'s text and
+ * start time, and appends it to {@code ulElement}.
+ */
+function appendTextToList(transcriptLine, ulElement) {
+  const liElement = document.createElement('li');
+  const infoDivElement = document.createElement('div');
+
+  appendPTagToContainer(transcriptLine.start, infoDivElement);
+  liElement.appendChild(infoDivElement);
+  appendPTagToContainer(transcriptLine.content, liElement);
+  liElement.appendChild(document.createElement('hr'));
+  ulElement.appendChild(liElement);
+}
+
+/**
+ * Creates a <p> tag to store the given {@code text} inside the
+ * {@code container} and returns the <p> tag using the given text.
+ */
+function appendPTagToContainer(text, container) {
+  const pTag = document.createElement('p');
+  pTag.innerText = text;
+  container.appendChild(pTag);
+  return pTag;
```

Comment:
Is there a reason why this is returned, I don't see the return being used in the caller

Reply:
I returned it so I can update the classes for the p tag in a later pull request.

I updated the method name to be `appendAndReturnPTagToContainer` to make it more clear that the p tag is returned. Thanks!

Reply:
I don't think we need to put return in the method name, reader could just look for a return statement

Reply:
Good point. Removed.